### PR TITLE
imp: team datasource add team name option

### PR DESCRIPTION
**Pull request by gmlp**
## Description
Improves team Datasource 
 - add team name option

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments

### CODE REVIEW

File: website/docs/d/team.html.markdown [hunk 1]
```diff
@@ -25,17 +25,27 @@ data "mongodbatlas_teams" "test" {
...
 
+
 ## Argument Reference
 
 * `org_id` - (Required) The unique identifier for the organization you want to associate the team with.
-* `team_id` - (Required) The unique identifier for the team.
+* `team_id` - (Optional) The unique identifier for the team.
+* `name` - (Optional) The team name.
+
+~> **IMPORTANT:** Either `team_id` or `name` must be configurated.
```
Comment (themantissa) [inline website/docs/d/team.html.markdown]: just spelling typo - "must be configured"

**Review by themantissa — CHANGES_REQUESTED** [review]
One typo noted in comment (and tests failing), otherwise LGTM so can approved once fixed!

File: mongodbatlas/data_source_mongodbatlas_team.go [hunk 2]
```diff
@@ -18,12 +19,14 @@ func dataSourceMongoDBAtlasTeam() *schema.Resource {
...
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"team_id"},
```
Comment (PacoDw) [inline mongodbatlas/data_source_mongodbatlas_team.go]: Maybe we can put them as a computed as well to set them and have/keep both in the TF state when one is chosen.

File: mongodbatlas/data_source_mongodbatlas_team_test.go [hunk 3]
```diff
@@ -52,5 +52,10 @@ func testAccDataSourceMongoDBAtlasTeamConfig(orgID, name, username string) strin
 			org_id     = mongodbatlas_teams.test.org_id
 			team_id    = mongodbatlas_teams.test.team_id
 		}
+
+		data "mongodbatlas_teams" "test2" {
+			org_id     = mongodbatlas_teams.test.org_id
+			name    = mongodbatlas_teams.test.name
+		}
```
Comment (PacoDw) [inline mongodbatlas/data_source_mongodbatlas_team_test.go:100]: I think is better to have another test case with this specific TF configuration to test separately

File: website/docs/d/team.html.markdown [hunk 4]
```diff
@@ -25,17 +25,27 @@ data "mongodbatlas_teams" "test" {
 	org_id     = mongodbatlas_teams.test.org_id
 	team_id    = mongodbatlas_teams.test.team_id
 }
+
+data "mongodbatlas_teams" "test2" {
+	org_id     = mongodbatlas_teams.test.org_id
+	name       = mongodbatlas_teams.test.name
+}
```
Comment (PacoDw) [inline website/docs/d/team.html.markdown:41]: The same that above comment, create a separate example using this case

File: website/docs/d/team.html.markdown [hunk 5]
```diff
@@ -25,17 +25,27 @@ data "mongodbatlas_teams" "test" {
...
+
+~> **IMPORTANT:** Either `team_id` or `name` must be configurated.
 
 
 ## Attributes Reference
 
 In addition to all arguments above, the following attributes are exported:
+
 * `id` -	The Terraform's unique identifier used internally for state management.
 * `name` -  The name of the team you want to create.
```
Comment (PacoDw) [inline website/docs/d/team.html.markdown:60]: Note that name is put here has a computed but your schema says that it's not, as we will change the `name` and `team_id` as computed, we need to put `team_id` here

**Review by PacoDw — CHANGES_REQUESTED** [review]

**Review by PacoDw — APPROVED** [review]

**Review by themantissa — APPROVED** [review]
LGTM!

**Pull request merged**